source/common/config/new_grpc_mux_impl.cc

```diff
@@ -153,13 +155,28 @@ GrpcMuxWatchPtr NewGrpcMuxImpl::addWatch(const std::string& type_url,
 // subscription will enqueue and attempt to send an appropriate discovery request.
 void NewGrpcMuxImpl::updateWatch(const std::string& type_url, Watch* watch,
                                  const std::set<std::string>& resources,
-                                 bool creating_namespace_watch) {
+                                 const bool creating_namespace_watch) {
   ASSERT(watch != nullptr);
   auto sub = subscriptions_.find(type_url);
   RELEASE_ASSERT(sub != subscriptions_.end(),
                  fmt::format("Watch of {} has no subscription to update.", type_url));
-  auto added_removed = sub->second->watch_map_.updateWatchInterest(watch, resources);
-  if (creating_namespace_watch) {
+  // If this is a glob collection subscription, we need to compute actual context parameters.
+  std::set<std::string> xdstp_resources;
+  if (resources.size() == 1 && XdsResourceIdentifier::hasXdsTpScheme(*resources.begin())) {
+    auto resource = XdsResourceIdentifier::decodeUrn(*resources.begin());
+    // We only know how to deal with glob collections right now.
+    if (absl::EndsWith(resource.id(), "/*")) {
+      auto encoded_context = XdsContextParams::encodeResource(
+          local_info_.contextProvider().nodeContext(), resource.context(), {}, {});
+      resource.mutable_context()->CopyFrom(encoded_context);
+      XdsResourceIdentifier::EncodeOptions encode_options;
+      encode_options.sort_context_params_ = true;
+      xdstp_resources.insert(XdsResourceIdentifier::encodeUrn(resource, encode_options));
```

Comment:
Currently, yes, context params can be updated only from the client.  I do have a proposal in #14428 to allow certain context params to be specified by the server, but I'm still in the process of getting consensus on that proposal.

Reply:
@markdroth if update from the Server side is allowed, how would the Client know about it? as the encoding will map the response with updated context parameters to nowhere? (since we are converting the whole resource name to an encoded string).

Reply:
The current proposal is to have a separate resource type that the server would use to send the dynamic parameters to the client, and the client would then use those parameters to update the resource names it is subscribing to.  For more info, see the proposal doc linked to in #14428.

Reply:
Right now, this doesn't support dynamic context params. This will be added in a followup PR, comment added.